### PR TITLE
Convert even more k8s tests to unit tests

### PR DESCRIPTION
**Pull request by jrockway**
I forgot we have //go:build k8s throughout the codebase.  This converts a bunch more of those to integration tests.

Testpachd can now run the S3 gateway and provide the tests with a fake k8s client (which is good enough for the secrets tests).

Jira: [MLDM-36]

[MLDM-36]: https://hpe-aiatscale.atlassian.net/browse/MLDM-36?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/internal/pachd/setup.go [hunk 1]
```diff
@@ -212,7 +212,7 @@ func initAuthServer(out *auth.APIServer, env func() authserver.Env) setupStep {
 		Fn: func(ctx context.Context) error {
 			apiServer, err := authserver.NewAuthServer(
 				env(),
-				false, false, false,
+				false, false, true,
```
[the remark this answers is not in the document]
Comment (robert-uhl) [reply src/internal/pachd/setup.go:215]: That third consecutive boolean (…) is `watchesEnabled`. It looks like it was originally meant to only be false in sidecars (perhaps a better, inverted name would have been ‘sidecarMode’). It stands to reason that it always ought to be true, and maybe this was a long-standing bug?

File: src/internal/pachd/testpachd.go [hunk 2]
```diff
@@ -65,6 +69,51 @@ func ActivateAuthOption(rootToken string) TestPachdOption {
...
+// WithS3Server runs the S3 gateway and sets the provided string pointer to its address.
+func WithS3Server(t *testing.T, addr *string) TestPachdOption {
+	l := testutil.Listen(t)
+	var server *http.Server
+	return TestPachdOption{
+		OnReady: func(ctx context.Context, full *Full) error {
+			full.s3Srv.Addr = l.Addr().String()
+			*addr = l.Addr().String()
+			server = full.s3Srv.Server
+			go full.s3Srv.Serve(l) //nolint:errcheck
```
Comment (robert-uhl) [inline src/internal/pachd/testpachd.go:91]: If you really, really wanted to you could do `go func() { doneCh <- full.s3Srv.Serve(l) }` and then read it down in the cleanup …

File: src/internal/pachd/setup.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (robert-uhl) [reply src/internal/pachd/setup.go:215]: It’s probably still broken for sidecars (and anyone else who disables watches) though, which should be a follow-on.

[the remark this answers is not in the document]
Comment (jrockway) [reply src/internal/pachd/setup.go:215]: I'm guessing that in the sidecars, the object is used for features other than being an RPC server, so nobody can call ModifyRoleBindings and thus avoid the crash.  Not ideal though.

**Comment by FahadBSyed** [comment]
It seems like the bazel-privileged-tests target is failing to build, circle seems like its getting a 255 error code.

**Comment by jrockway** [comment]
> It seems like the bazel-privileged-tests target is failing to build, circle seems like its getting a 255 error code.

Yeah I'm working on it.  It's because the docker containers needed by the tests need to be started before the test starts (for privileged tests) and this is the first test that needs minio, so that support is being added.

**Comment by jrockway** [comment]
> It seems like the bazel-privileged-tests target is failing to build, circle seems like its getting a 255 error code.

OK, I finally sorted this out.  This should make the non-privileged tests more reliable as well.

**Review by FahadBSyed — APPROVED** [review]

**Pull request merged**